TGServerService/Compiler.cs

```diff
@@ -383,6 +383,13 @@ void CompileImpl()
 					}
 				}
 
+				if (!PrecompileHook())
+				{
+					lastCompilerError = "The precompile hook failed";
+					compilerCurrentStatus = TGCompilerStatus.Initialized;   //still fairly valid
+					return;
```

Comment:
One more thing, both the pre and post compile hooks should write to the log e.g.
```
TGServerService.WriteWarning("Precompile hook failed!", TGServerService.EventID.DMCompileError);
```